### PR TITLE
Revised Patron Tile (UI) with flexy height

### PR DESCRIPTION
**Pull request by nsharma123**
https://webarchive.jira.com/browse/WEBDEV-5625

In this revision, I followed following coding structure, looking to have more suggestion on below structure.

- render()
- template getters
- other component lifecycle methods like update()
- other methods
- css

### CODE REVIEW

File: src/collection-browser.ts [hunk 1]
```diff
@@ -1801,14 +1801,6 @@ export class CollectionBrowser
         18rem
       );
       --infiniteScrollerCellMaxWidth: var(--collectionBrowserCellMaxWidth, 1fr);
-      --infiniteScrollerCellMinHeight: var(
```
Comment (dualcnhq) [inline src/collection-browser.ts:1804]: 👍

Comment (dualcnhq) [reply src/collection-browser.ts:1804]: this is now merged to main branch, feel free to rebase it

File: src/tiles/grid/account-tile.ts [hunk 2]
```diff
@@ -1,123 +1,74 @@
...
   render() {
     return html`
-      <div class="account-tile-main">
-        <div id="title">
-          <h1 class="truncated">${this.model?.identifier}</h1>
-        </div>
-
-        <div class="account-info">
-          <div id="avatar-info">
-            <img
```
Comment (nsharma123) [inline src/tiles/grid/account-tile.ts:20]: replaced this with <image-block> components

File: src/tiles/grid/account-tile.ts [hunk 3]
```diff
@@ -1,123 +1,74 @@
...
+  }
 
-      #avatar-info {
-        margin-top: 5px;
-        margin-bottom: 5px;
-        display: flex;
-        align-items: center;
-        justify-content: center;
-      }
+  private get getTitle() {
```
Comment (nsharma123) [inline src/tiles/grid/account-tile.ts:39]: getter function for different blocks

File: src/tiles/grid/account-tile.ts [hunk 4]
```diff
@@ -1,123 +1,74 @@
...
+    return html`<tile-stats
+      .mediatype=${this.model?.mediatype}
+      .itemCount=${this.model?.itemCount}
+      .favCount=${this.model?.favCount}
+      .commentCount=${this.model?.commentCount}
+    >
+    </tile-stats>`;
+  }
 
-      .truncated {
```
Comment (nsharma123) [inline src/tiles/grid/account-tile.ts:109]: moved `.truncated` class in shared styles.

File: src/tiles/grid/styles/tile-grid-shared-styles.ts [hunk 5]
```diff
@@ -0,0 +1,98 @@
+import { css } from 'lit';
```
Comment (nsharma123) [inline src/tiles/grid/styles/tile-grid-shared-styles.ts:1]: this shared style being used for tile system (item/patron/collection/etc...) only from now.

File: src/tiles/image-block.ts [hunk 6]
```diff
@@ -81,12 +81,14 @@ export class ImageBlock extends LitElement {
...
   static get styles(): CSSResultGroup {
+    const imageBackgroundColor = css`var(--imageBackgroundColor, #f1f1f4)`;
+
     return css`
       div {
         display: flex;
         justify-content: center;
         position: relative;
-        background-color: #f1f1f4;
+        background-color: ${imageBackgroundColor};
```
Comment (nsharma123) [inline src/tiles/image-block.ts:91]: background color difference for different tiles. Like patron tiles and item tiles using different background color for image section.

File: src/tiles/item-image.ts [hunk 7]
```diff
@@ -81,9 +81,21 @@ export class ItemImage extends LitElement {
...
       blur: toBlur || false,
       waveform: this.isWaveform,
+      radius50: this.itemImageRadius,
     };
   }
 
+  /**
+   * Helper function to determine if border radius is needed
+   */
+  private get itemImageRadius() {
```
Comment (nsharma123) [inline src/tiles/item-image.ts:91]: apply 50% radius on patron image

File: src/tiles/grid/styles/tile-grid-shared-styles.ts [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+
+/**
+ * Base tile styles
+ */
+
+const tileCornerRadius = css`var(--collectionTileCornerRadius, 4px)`;
+const tileBackgroundColor = css`var(--TileBackgroundColor, #ffffff)`;
```
Comment (latonv) [inline src/tiles/grid/styles/tile-grid-shared-styles.ts:8]: Can we make this `--tileBackgroundColor` (lowercase t) for consistency?

Comment (dualcnhq) [reply src/tiles/grid/styles/tile-grid-shared-styles.ts:8]: +1

File: src/tiles/grid/styles/tile-grid-shared-styles.ts (hunk 5 above)
Comment (latonv) [reply src/tiles/grid/styles/tile-grid-shared-styles.ts:1]: Nice! I like that there's a single source of truth for some of these shared tile styles now.

File: test/tiles/grid/account-tile.test.ts [hunk 9]
```diff
@@ -0,0 +1,95 @@
+/* eslint-disable import/no-duplicates */
```
Comment (dualcnhq) [inline test/tiles/grid/account-tile.test.ts:1]: thank you for adding test 💯

File: src/tiles/grid/account-tile.ts (hunk 3 above)
Comment (latonv) [reply src/tiles/grid/account-tile.ts:39]: I like that these have been pulled outside of the render method. Could we name the getters more like `avatarTemplate`, `titleTemplate` etc, to stay consistent with how some of our other components name template getters? I think that more clearly conveys the purpose.

Comment (nsharma123) [reply src/tiles/grid/account-tile.ts:39]: Agreed with you. Updating these as well. Thanks!

File: test/tiles/grid/account-tile.test.ts [hunk 10]
```diff
@@ -9,18 +9,18 @@ describe('Account Tile', () => {
   it('should render initial component', async () => {
     const el = await fixture<AccountTile>(html`<account-tile></account-tile>`);
 
-    const itemInfo = el.shadowRoot?.querySelector('.account-info');
-    const itemAvatar = el.shadowRoot?.querySelector('#avatar-info');
-    const itemArchivist = el.shadowRoot?.querySelector('#archivist-since');
+    const itemInfo = el.shadowRoot?.querySelector('.item-info');
+    const ItemImage = el.shadowRoot?.querySelector('image-block');
```
Comment (latonv) [inline test/tiles/grid/account-tile.test.ts:13]: Minor capitalization inconsistency on `ItemImage`

**Review by dualcnhq — APPROVED** [review]

**Pull request merged**